torch/distributions/transforms.py

```diff
@@ -600,6 +601,26 @@ def log_abs_det_jacobian(self, x, y):
         return -F.softplus(-x) - F.softplus(x)
 
 
+class SoftplusTransform(Transform):
+    r"""
+    Transform via the mapping :math:`\text{Softplus}(x) = \log(1 + \exp(x))`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def __eq__(self, other):
+        return isinstance(other, SoftplusTransform)
+
+    def _call(self, x):
+        return x.exp().log1p()
+
+    def _inverse(self, y):
+        return y.expm1().log()
+
+    def log_abs_det_jacobian(self, x, y):
+        return -(-x).exp().log1p()
```

Comment:
A more stable formula would be: `-softplus(-x)`

Reply:
Same here - https://github.com/pytorch/pytorch/pull/52300#discussion_r579817195